python/beeai_framework/emitter/emitter.py

```diff
@@ -182,7 +185,10 @@ async def invoke(self, data: Any, event: EventMeta) -> None:
                 self.listeners.remove(listener)
 
             async def run(ln: Listener = listener) -> Any:
-                return await ln.callback(data, event)
+                if inspect.iscoroutinefunction(ln.callback):
+                    return await ln.callback(data, event)
+                else:
+                    return ln.callback(data, event)
```

Comment:
could be extracted to another PR
cc @ajbozarth

Reply:
I was about to argue against this because I thought `ln.callback` had to be async, but upon review it actually doesn't. This would also allow us to remove async from a bunch of the callback defs. I can also include this in a separate cleanup PR. I'll have one open by EOD for review before release tomorrow